packages/ubuntu_wsl_setup/lib/services/named_event.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'dart:collection';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+/// Exposes a [future] that completes when a platform event named [name] is set.
+class NamedEvent {
+  /// The event name as known by the platform, UTF-8 encoded.
+  String name;
+  final _completer = Completer<void>();
+
+  /// Completes when the platform event is set.
+  Future<void> get future => _completer.future;
+  NamedEvent(this.name) {
+    _NamedEventSingleton.instance.addListenerFor(
+      name: name,
+      listener: _completer.complete,
+    );
+  }
+}
+
+@visibleForTesting
+class NamedEventConstants {
+  static const channel = 'ubuntuWslSetupChannel';
+  static const onEventSet = 'onEventSet';
+  static const addListenerFor = 'addListenerFor';
+  static const argKey = 'eventName';
```

Comment:
we could get rid of this if we pass the event name directly

Reply:
Indeed